x/concentrated-liquidity/position_test.go

```diff
@@ -937,6 +937,208 @@ func (s *KeeperTestSuite) TestHasAnyPosition() {
+			} else if test.isUnlocking {
+				positionId, _, _, liquidity, _, concentratedLockId, err = s.App.ConcentratedLiquidityKeeper.CreateFullRangePositionUnlocking(s.Ctx, clPool, defaultAddress, defaultPositionCoins, test.remainingLockDuration)
+			} else {
+				positionId, _, _, liquidity, _, err = s.App.ConcentratedLiquidityKeeper.CreateFullRangePosition(s.Ctx, clPool, defaultAddress, defaultPositionCoins)
+			}
+
+			s.Require().NoError(err)
+
+			// Check position
+			_, err = s.App.ConcentratedLiquidityKeeper.GetPosition(s.Ctx, positionId)
+			s.Require().NoError(err)
+
+			// Check lock
+			if test.isLocked || test.isUnlocking {
+				concentratedLock, err := s.App.LockupKeeper.GetLockByID(s.Ctx, concentratedLockId)
+				s.Require().NoError(err)
+				s.Require().Equal(liquidity.TruncateInt().String(), concentratedLock.Coins[0].Amount.String())
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestMintSharesLockAndUpdate() {
+	defaultAddress := s.TestAccs[0]
+	defaultPositionCoins := sdk.NewCoins(DefaultCoin0, DefaultCoin1)
+
+	tests := []struct {
+		name                    string
+		owner                   sdk.AccAddress
+		remainingLockDuration   time.Duration
+		createFullRangePosition bool
+	}{
+		{
+			name:                    "2 week lock",
+			owner:                   defaultAddress,
+			createFullRangePosition: true,
+			remainingLockDuration:   24 * time.Hour * 14,
+		},
+		{
+			name:                    "1 day lock",
+			owner:                   defaultAddress,
+			createFullRangePosition: true,
+			remainingLockDuration:   24 * time.Hour,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		s.Run(test.name, func() {
+			// Init suite for each test.
+			s.Setup()
+			s.Ctx = s.Ctx.WithBlockTime(DefaultJoinTime)
+
+			// Create a default CL pools
+			clPool := s.PrepareConcentratedPool()
+
+			// Fund the owner account
+			s.FundAcc(test.owner, defaultPositionCoins)
+
+			// Create a position
+			positionId := uint64(0)
+			liquidity := sdk.ZeroDec()
+			if test.createFullRangePosition {
+				var err error
+				positionId, _, _, liquidity, _, err = s.App.ConcentratedLiquidityKeeper.CreateFullRangePosition(s.Ctx, clPool, test.owner, defaultPositionCoins)
+				s.Require().NoError(err)
+			}
+
+			// System under test
+			concentratedLockId, underlyingLiquidityTokenized, err := s.App.ConcentratedLiquidityKeeper.MintSharesLockAndUpdate(s.Ctx, clPool, positionId, test.owner, test.remainingLockDuration, liquidity)
+			s.Require().NoError(err)
+
+			// Check lock
+			concentratedLock, err := s.App.LockupKeeper.GetLockByID(s.Ctx, concentratedLockId)
+			s.Require().NoError(err)
+			s.Require().Equal(underlyingLiquidityTokenized[0].Amount.String(), concentratedLock.Coins[0].Amount.String())
+			s.Require().Equal(test.remainingLockDuration, concentratedLock.Duration)
```

Comment:
Would like to see the following checks here:
1. Balances update
2. position id to lock index update
3. Validate that supply increased by this cl token (the fact of minting)

Reply:
test extended here https://github.com/osmosis-labs/osmosis/pull/4883/commits/7dd0b55bee09a0a36621a52cd47d79ae1b55fa6f